### PR TITLE
Migration to HUGO

### PR DESCRIPTION
**Pull request by kritika-singh3**
### Summary

1. Added rake task to use html-proofer to test external links - linked it with build task.
2. Fixed broken links identified via the above task.
3. For now, ignored the links with 'svn' in it.

P.S.: List of links updated:
https://docs.google.com/spreadsheets/d/1ItYXzYIXeeUmmMLceFkDBcI3r46pCIZRqYmsyvCzLD8/edit?usp=sharing

### CODE REVIEW

File: content/configuration/configuration_reference.md [hunk 1]
```diff
@@ -1039,7 +1039,7 @@ The config repo `<p4>` material element specifies the location of your code base
 | username | No | Perforce username to use. |
 | password | No | Password for the specified user. |
 | encryptedPassword | No | Encrypted Password for the specified user. |
-| useTickets | No | Set to true to work with perforce tickets. Go will do a p4 login using the supplied password before each command. We recommend that you make your user a part of a p4 group, and set the ticket timeout to unlimited as described [here](http://www.perforce.com/perforce/doc.current/manuals/cmdref/login.html). |
+| useTickets | No | Set to true to work with perforce tickets. Go will do a p4 login using the supplied password before each command. We recommend that you make your user a part of a p4 group, and set the ticket timeout to unlimited as described [here](https://www.perforce.com/manuals/cmdref/Content/CmdRef/p4_login.html). |
```
Comment (SanjanaB) [inline content/configuration/configuration_reference.md]: How did u derive this url? should we double check with someone?

File: content/extension_points/_index.md [hunk 2]
```diff
@@ -20,29 +20,29 @@ Please go through [Go Plugin User Guide](plugin_user_guide.html) before using Go
 
 # Package Repository Extension
 
-Please follow [link](package_repository_extension.html) for package material extension point details.
+Please follow [link](extension_points/package_repository_extension.html) for package material extension point details.
```
Comment (SanjanaB) [inline content/extension_points/_index.md]: it is referenced from same directory. this should not be needed.

File: content/extension_points/yum_repository_poller.md [hunk 3]
```diff
@@ -27,7 +27,7 @@ Using the [Yum plugin for Nexus](https://github.com/sonatype/nexus-yum-plugin),
 
 ### Repository definition
 
-Repo URL must be a valid http, https or file URL. This plugin looks for the presence of **\$REPO\_URL/repodata/[repomd.xml](http://createrepo.baseurl.org/wiki)** to ascertain validity. Basic authentication (user:password@domain/path) is supported for http and https repositories.
+Repo URL must be a valid http, https or file URL. This plugin looks for the presence of **\$REPO\_URL/repodata/[repomd.xml](http://createrepo.baseurl.org)** to ascertain validity. Basic authentication (user:password@domain/path) is supported for http and https repositories.
```
Comment (SanjanaB) [inline content/extension_points/yum_repository_poller.md]: Can we check with someone on this url change too?

File: content/configuration/configuration_reference.md (hunk 1 above)
Comment (kritika-singh3) [reply content/configuration/configuration_reference.md]: Took @ketan help in resolving all the major URLs

File: content/extension_points/_index.md (hunk 2 above)
Comment (kritika-singh3) [reply content/extension_points/_index.md]: The other files are bundled inside a folder and the index resides outside it after the build. Hence, the need to reference it explicitly.

File: content/integration/_index.md [hunk 4]
```diff
@@ -12,7 +12,7 @@ GoCD allows you to link your commit messages with bug tracking and story managem
 
 The following shows an example of tracking cards on Mingle by card numbers starting with a "\#" (e.g: \#4618):
 
-![Configure tracking tool](../images/pipeline_tracking_tool.png)
+![Configure tracking tool](./images/pipeline_tracking_tool.png)
```
Comment (SanjanaB) [inline content/integration/_index.md]: should be a relative url. images are one level above the integration directory.

Comment (kritika-singh3) [reply content/integration/_index.md]: The integration folder has only one _index file. Hence, a separate folder won't be created but only an index.html which resides in the same folder as images folder.

Comment (SanjanaB) [reply content/integration/_index.md]: you mean integration.html right?

Comment (kritika-singh3) [reply content/integration/_index.md]: yes

**Comment by SanjanaB** [comment]
Is the static_checks task included part of build pipeline?

**Comment by SanjanaB** [comment]
Also upload.rake file is checked in but never used. I am introducing this in my commit. You can remove it if your commit is not using the file.

**Comment by SanjanaB** [comment]
RUN_EXTERNAL_CHECKS can be set in the build.gocd.groovy file so that the pipeline is set with this env variable.

**Pull request merged**